e2e-tests/sidebar.spec.ts

```diff
@@ -25,13 +25,14 @@ test('favorite item and recent item test', async ({ page }) => {
 
   // remove fav
   await page.click('.ui__dropdown-trigger')
-  await page.click(':nth-match(.ui__dropdown-trigger .dropdown-wrapper a, 1)')
+  await page.locator("text=Unfavorite page").click()
   await page.waitForTimeout(1000)
   favs = await page.$$('.favorite-item a')
   expect(favs.length).toEqual(previous_fav_count)
 
   // click from fav page
   await page.click(':nth-match(.recent-item a, 2)')
+  await page.waitForTimeout(1000)
   expect(await page.innerText('.page-title .title')).toBe(another_page_name)
```

Comment:
```suggestion
  await  expect(page.innerText('.page-title .title')).toBe(another_page_name)
```
instead of waitForTimeout

Reply:
Seems not working?

Reply:
hmm, I made a mistake.
`await expect(page.locator('....'))toHaveText`
Never mind, `waitForTimeout` is OK.

Reply:
ref: https://playwright.dev/docs/test-assertions#locator-assertions-to-have-text

Reply:
Locator assertions are useful to eliminate `waitForTimeout`s.